src/main/scala/rps/GameController.scala

```diff
@@ -7,30 +7,26 @@ import wiro.Config
 import wiro.server.akkaHttp._
 import wiro.server.akkaHttp.FailSupport._
 import io.buildo.enumero.{CaseEnumIndex, CaseEnumSerialization}
+import HashId._
 
 @path("rps")
-trait GameController {
+trait GameController  {
  @command
-  def play(userMove:Move): Future[Either[Throwable,Unit]]
+  def play(userMove:Move): Future[Either[Throwable,IdType]]
 
   @query
-  def result(): Future[Either[Throwable,Response]]
+  def result(id: IdType): Future[Either[Throwable,Response]]
 }
 
-
-class GameControllerImpl(gs : GameService)(implicit ec: ExecutionContext) extends GameController {
-  override def play(userMove:Move) : Future[Either[Throwable,Unit]] = Future {
+class GameControllerImpl(gs: GameService)(implicit ec: ExecutionContext) extends GameController {
+  override def play(userMove:Move) : Future[Either[Throwable,IdType]] = Future {
```

Comment:
```suggestion
  override def play(userMove:Move) : Future[Either[Throwable,IdType]] = Future.successful(Right(gs.play(userMove)))
```

Reply:
this will avoid to create a thread, it just lifts your result in a completed future